src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -327,5 +329,6 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
             datasetId,
             IamResourceType.DATASET,
             "A snapshot was created from this dataset."));
+    addStep(new CreateSnapshotPolicyStep(policyService, sourceDataset.isSecureMonitoringEnabled()));
```

Comment:
If the state of the conditional is known at flight creation time, you can avoid the need to check it in the step by making the step addition conditional.
```suggestion
    if (sourceDataset.isSecureMonitoringEnabled) {
      addStep(new CreateSnapshotPolicyStep(policyService));
    }
```

Reply:
I left the logic within the step because later on we will need to check other properties on the snapshot, like whether it has an auth domain/allow-list or a region constraint

Reply:
> later on we will need to check other properties on the snapshot

@samanehsan , do you mean source dataset here instead of snapshot?  If that's the case, it may make more sense to pass the whole source dataset to the step if you'll need to pull out more individual fields in the future.

Reply:
I'm also curious about the placement of this step -- I see that it's taking place after the snapshot has been unlocked and the snapshot creation has been journaled.  Is that intentional?

Reply:
> do you mean source dataset here instead of snapshot? If that's the case, it may make more sense to pass the whole source dataset to the step if you'll need to pull out more individual fields in the future.

I think we'll need both! Right now we don't have a way for users to specify that a dataset/snapshot must remain in a certain geographic region (which would correspond to the `region-constraint` policy), but that would get set on the dataset-level and inherited by the snapshot (just like the region is).

The auth-domain/allow-list of users (which would correspond to the `group-constraint` policy) could differ between snapshots of the same dataset, so that could be part of the snapshot create request (or perhaps a different endpoint to update the existing snapshot).

I'm inclined to leave this as-is and then make the change when we support the other policies if that's okay with you!

Reply:
That makes good sense!